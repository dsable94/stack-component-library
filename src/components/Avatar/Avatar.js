import React from "react";
import { StyledAvatar, StyledAvatarText, AvatarTitle, AvatarSubTitle } from './Avatar.styled';
import Typography from "../Typography/Typography.styled";

const Avatar = (props) => {
  const { size, src, alt, title, subTitle, titleFont, subTitleFont, color, isBold } = props;
  return (
    <StyledAvatarText {...props}>
      <StyledAvatar size={size} src={src} alt={alt} >
      </StyledAvatar>
     { title && <AvatarTitle titleFont={titleFont} isBold={isBold} >
        {title}
       </AvatarTitle>}
      { subTitle && <AvatarSubTitle subTitleFont={subTitleFont}>
        {subTitle}
      </AvatarSubTitle> }
    </StyledAvatarText>
  );
}

export default Avatar;