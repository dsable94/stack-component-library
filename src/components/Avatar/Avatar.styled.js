import styled, {css} from "styled-components";

export const StyledAvatarText = styled.div`

`
export const AvatarTitle = styled.div`
${props=> props.titleFont ? css ` font-size: ${props.titleFont};` : css ``}
${props=> props.isBold ? css ` font-weight: ${props.isBold};` : css ``}
${props=> props.titleColor ? css ` color: ${props.titleColor};` : css ``}

`
export const AvatarSubTitle = styled.div`
${props=> props.subTitleFont ? css ` font-size: ${props.subTitleFont};` : css ``}
${props=> props.subTitleColor ? css ` color: ${props.subTitleColor};` : css ``}

`

export const StyledAvatar = styled.img`
  vertical-align: middle;
  border-radius: 9999px;
  margin-bottom: 10px;
  ${props =>
  props.size === 'sm'
  ? css `
  height: 35px;
  width: 35px;
  `
  : props.size === 'lg'
  ? css `
  height: 75px;
  width: 75px;
  `
  : css `
  height: 50px;
  width: 50px;
  `
  }
  
`