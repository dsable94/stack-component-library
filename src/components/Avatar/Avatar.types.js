import React from "react";
import PropTypes from "prop-types";

const AvatarTypes= () => <>AvatarTypes</>;

AvatarTypes.propTypes={
  src: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['sm','md', 'lg']),
  alt: PropTypes.string,
  title: PropTypes.string,
  subTitle: PropTypes.string,
  titleFont: PropTypes.string,
  subTitleFont: PropTypes.string,
  isBold: PropTypes.string,
};

AvatarTypes.defaultProps = {};

export default AvatarTypes;