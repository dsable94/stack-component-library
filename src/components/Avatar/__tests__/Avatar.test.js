
import React from 'react';
import {  shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import Avatar from '../Avatar';

// describe('***--- Avatar Component --- ***', () => {
//   it('test', () => {
//     expect(1).toEqual(1);
//   });
// });

describe('Connections Table', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <Avatar />,
    );
  });
  // beforeEach(() => {
  //   wrapper = shallow(
  //     <Avatar />)
  //   //   <CmsContext.Provider value={cmsContextValue}>
  //   //     <FeatureComponent {...props} />
  //   //   </CmsContext.Provider>,
  //   // ).dive().dive().dive();
  // });

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
    console.log('-----------', wrapper.debug());
  });

//   it('Simulate mutation onClick on Button', async () => {
//     wrapper.find('#header').simulate('click');
//   });
});
