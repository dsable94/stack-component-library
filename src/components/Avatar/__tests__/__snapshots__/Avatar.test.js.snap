// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connections Table should render properly 1`] = `<Avatarstyled__StyledAvatar />`;
